src/ucs/sys/sock.c

```diff
@@ -391,6 +391,13 @@ ucs_status_t ucs_socket_server_init(const struct sockaddr *saddr, socklen_t sock
         if (status != UCS_OK) {
             goto err_close_socket;
         }
+#ifdef SO_REUSEPORT
+        status = ucs_socket_setopt(fd, SOL_SOCKET, SO_REUSEPORT,
```

Comment:
if I'm not mistaken `so_reuseaddr` should be enough
if it fails on TCP CM - we can set UCX_REUSEADDR=y

Reply:
```UCX_TCP_CM_REUSEADDR=y``` doesn't help - as I can see process is killed and kernel takes some timeout to switch it into inactive state

Reply:
i see, thanks for checking

Reply:
btw, did you see on TCP or TCP CM?
if TCP, can force `reuse_addr=1` to check if the problem there

Reply:
```UCX_TCP_CM_REUSEADDR=y``` forces ```reuse_addr=1``` and it doesn't wirk for TCP - failed to init listener

Reply:
> `UCX_TCP_CM_REUSEADDR=y` forces `reuse_addr=1` and it doesn't wirk for TCP - failed to init listener

it forces for TCP CM
my idea to check for TCP transport, it passes `0` always to the function:
https://github.com/openucx/ucx/blob/787eba4df83c2d305147bf1c590fa228da1837ce/src/uct/tcp/tcp_iface.c#L397

Reply:
failure is on call https://github.com/openucx/ucx/blob/787eba4df83c2d305147bf1c590fa228da1837ce/src/uct/tcp/tcp_listener.c#L117

Reply:
I see, thanks!